mantidproject/mantid
# D16B commissioning

**Description of work.**

This PR implements recently received comments regarding D16B data reduction:
- [x] D16B has now two monitors in the scanned variables, and both are now loaded
- [x] D16B now certainly loads proper Monitor1 and Monitor2 from scanned_variables which was previously not ensured as it did not depend on the actual search of the scanned names but the position of data in the table (index)  
- [x] `SimpleScanViewer` now fits the X and Y position of peak in each ROI, instead of centre-of-mass calculation
- [x] Container and absorber corrections are now connected properly in `SANSILLParameterScan` algorithm
- [x] Container measurement can now be a scan in `SANSILLReduction` v2

**To test:**

1. Load old D16 data omega scan from the unit test suite, e.g. 025786.nxs. It should load correctly and contain two monitors with the second one being not filled but with correct binning on X axis.
2. Load new D16B data omega scan, e.g. 066321.nxs. It should also have two monitors.
3.  For the SimpleScanViewer tests:
    1. Open workbench.
    2. Open SimpleScanViewer interface by navigating through: Interfaces->ILL->SimpleScanViewer. 
    3. Browse to this file: [d16_exp_9-13-1035.nxs](https://github.com/scientific-software/mantill/uploads/d274147e001304088be7548c51657389/d16_exp_9-13-1035.nxs)
    4. The reduction should pass without issues and a new window should appear.
    5. Test that when you draw a ROI on a visible peak, the fitted position correspond to a reasonable outcome.
    6. Export peak table (by pressing p) and observe parameters there.
    7. Try to interpolate the ROIs, each interpolated ROI should find a peak if it was contained in it.
    8. Close the interface.
    9. Open it again and repeat points 3-7 by starting with modified parameters in dialog that appears when you press Advanced button.

Fixes #35092 . 


Release notes and update documentation updated.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
